rapids-cmake/cpm/detail/get_proprietary_binary.cmake

```diff
@@ -0,0 +1,78 @@
+#=============================================================================
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+include_guard(GLOBAL)
+
+#[=======================================================================[.rst:
+get_proprietary_binary
+-------------------
+
+.. versionadded:: v22.06.00
+
+Download the associated proprietary binary for the given project based on
+the current CPU target architecture ( x86_64, aarch64, etc )
+
+ .. note::
+  if override => the proprietary entry only in the override will be evaluated
+  if no override => the proprietary entry only in the default will be evaluated
+
+
+#]=======================================================================]
+function(rapids_cpm_get_proprietary_binary package_name version)
+
+  get_property(override_json_data GLOBAL PROPERTY rapids_cpm_${package_name}_override_json)
+  # need to search the `proprietary_binary` dictonary for a key with the same name as
+  # lower_case(`CMAKE_SYSTEM_PROCESSOR-CMAKE_SYSTEM_NAME`).
+  set(key "${CMAKE_SYSTEM_PROCESSOR}-${CMAKE_SYSTEM_NAME}")
+  string(TOLOWER ${key} key)
+
+  if(override_json_data)
+    string(JSON value ERROR_VARIABLE have_error GET "${override_json_data}" "proprietary_binary"
+           "${key}")
+  else()
+    get_property(json_data GLOBAL PROPERTY rapids_cpm_${package_name}_json)
+    string(JSON value ERROR_VARIABLE have_error GET "${json_data}" "proprietary_binary" "${key}")
+  endif()
+
+  if(have_error)
+    message(VERBOSE
+            "${package_name} requested usage of a proprietary_binary but none exist for ${CMAKE_SYSTEM_PROCESSOR}"
+    )
+    return()
+  endif()
+
+  # proprietary_binary now is the url that we need to download
+  set(proprietary_binary ${value})
```

Comment:
Why not set this variable directly in the `string` commands above? The message/return is fairly self-documenting that `proprietary_binary` may not be well-defined if `have_error` is `TRUE`.